com.adobe.Flash-Player-Projector.json

```diff
@@ -0,0 +1,69 @@
+{
+    "app-id" : "com.adobe.Flash-Player-Projector",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.28",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "flashplayer",
+    "separate-locales": false,
+    "tags": ["proprietary"],
+    "finish-args" : [
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+        /* Play sounds */
+        "--socket=pulseaudio"
+    ],
+    "modules" : [
+        {
+            "name" : "flashplayer",
+            "buildsystem" : "simple",
+            "build-commands" : [
+                "install -D -t /app/bin/ apply_extra",
+                "install flashplayer /app/bin",
+                "install -Dm644 -t /app/share/icons/ com.adobe.Flash-Player-Projector.svg",
+                "install -Dm644 -t /app/share/applications/ com.adobe.Flash-Player-Projector.desktop",
+                "install -Dm644 -t /app/share/metainfo/ com.adobe.Flash-Player-Projector.appdata.xml"
+            ],
+            "sources" : [
+                {
+                    "type" : "script",
+                    "dest-filename" : "flashplayer",
+                    "commands": [
+                       "exec /app/extra/bin/flashplayer.bin $@"
+                    ]
+                },
+                {
+                   "type" : "script",
+                   "dest-filename": "apply_extra",
+                   "commands": [
+                       "echo Linked from https://www.adobe.com/support/flashplayer/debug_downloads.html",
+                       "tar xvzf flash_player_sa_linux.x86_64.tar.gz",
+                       "rm -f flash_player_sa_linux.x86_64.tar.gz",
+                       "install -Dm755 flashplayer /app/extra/bin/flashplayer.bin"
+                   ]
+                },
+                {
+                    "type" : "extra-data",
+                    "filename" : "flash_player_sa_linux.x86_64.tar.gz",
+                    "only-arches": ["x86_64"],
+                    "url" : "https://fpdownload.macromedia.com/pub/flashplayer/updaters/30/flash_player_sa_linux.x86_64.tar.gz",
+                    "sha256" : "b9b2ef564ef8474aad6379bcc9f48894d5ec2ae479564201a40aaa0f70a1e1f7",
+                    "size" : 8501186
+                },
+                {
+                    "type" : "file",
+                    "url" : "https://helpx.adobe.com/content/dam/help/mnemonics/flashplayer_app_RGB.svg",
```

Comment:
If you don't use `extra-data`, the icon will be hosted on flathub.org and if that's okay legally, you might as well put it in the git repo.

Reply:
I'd rather have a minimum number of files in the repo itself, that way I don't need to figure out that it changed before getting warned about it.

Reply:
If you put it in the repo, it will never change and you can just concentrate on the extra-data.